test/functional/mempool_limit.py

```diff
@@ -5,70 +5,90 @@
-        assert_equal(self.nodes[0].getmempoolinfo()['minrelaytxfee'], Decimal('0.00001000'))
-        assert_greater_than(self.nodes[0].getmempoolinfo()['mempoolminfee'], Decimal('0.00001000'))
-
-        self.log.info('Create a mempool tx that will not pass mempoolminfee')
-        us0 = utxos.pop()
-        inputs = [{ "txid" : us0["txid"], "vout" : us0["vout"]}]
-        outputs = {self.nodes[0].getnewaddress() : 0.0001}
-        tx = self.nodes[0].createrawtransaction(inputs, outputs)
-        # specifically fund this tx with a fee < mempoolminfee, >= than minrelaytxfee
-        txF = self.nodes[0].fundrawtransaction(tx, {'feeRate': relayfee})
-        txFS = self.nodes[0].signrawtransactionwithwallet(txF['hex'])
-        assert_raises_rpc_error(-26, "mempool min fee not met", self.nodes[0].sendrawtransaction, txFS['hex'])
-
-if __name__ == '__main__':
+
+        node = self.nodes[0]
+        miniwallet = MiniWallet(node)
+        relayfee = node.getnetworkinfo()["relayfee"]
+
+        self.log.info("Check that mempoolminfee is minrelytxfee")
+        assert_equal(node.getmempoolinfo()["minrelaytxfee"], Decimal("0.00001000"))
+        assert_equal(node.getmempoolinfo()["mempoolminfee"], Decimal("0.00001000"))
+
+        # Generate 92 UTXOs to flood the mempool
+        # 1 to create a tx initially that will be evicted from the mempool later
+        # 90 with a fee rate much higher than the previous UTXO (3 batches of 30 with increasing fee rate)
+        # And 1 more to verify that this tx does not get added to the mempool with a fee rate less than the mempoolminfee
+        miniwallet.generate(1 + (3 * 30) + 1)
+
+        # Mine 99 blocks so that the UTXOs are allowed to be spent
+        node.generate(COINBASE_MATURITY - 1)
+
+        self.log.info("Create a mempool tx that will be evicted")
+        tx_to_be_evicted_id = miniwallet.send_self_transfer(
+            from_node=node, fee_rate=relayfee
+        )["txid"]
+
+        # Increase the tx fee rate massively now to give the next transactions a higher priority in the mempool
+        base_fee = relayfee * 1000
+
+        self.log.info("Fill up the mempool with txs with higher fee rate")
+        no_of_large_tx_created = 0
+        for batch_of_txid in range(3):
+            # Increment the tx fee rate gradually by a factor of (basee_fee) for each batch of 30 transactions
+            no_of_large_tx_created += miniwallet.create_large_transactions(
+                node, txouts, 30, (batch_of_txid + 1) * base_fee
+            )
+
+        self.log.info("The tx should be evicted by now")
+        # The number of transactions created should be greater than the ones present in the mempool
+        assert_greater_than(no_of_large_tx_created, len(node.getrawmempool()))
+        # Initial tx created should not be present in the mempool anymore as it had a lower fee rate
+        assert tx_to_be_evicted_id not in node.getrawmempool()
+
```

Comment:
In extension to the corresponding comment above, maybe we should add the confirmation test, something on the lines of:
```
txdata = node.gettransaction(tx_to_be_evicted)
assert txdata['confirmations'] == 0
```

Reply:
With respect to this and the above comment, `gettransaction(tx_to_be_evicted)` is a method that requires a wallet to be loaded, contradictory to the purpose of this PR. Hence, we're just verifying if it's evicted from the mempool!